tests/test_http_auth.py

```diff
@@ -203,64 +204,70 @@ def test_http_auth_scram_invalid_password(self):
-        self.assertEqual(
-            headers['Access-Control-Allow-Origin'],
-            'https://example.edgedb.com'
-        )
-        self.assertIn('GET', headers['Access-Control-Allow-Methods'])
-        self.assertIn('Authorization', headers['Access-Control-Allow-Headers'])
-        self.assertIn(
-            'WWW-Authenticate',
-            headers['Access-Control-Expose-Headers']
-        )
-        self.assertIn(
-            'Authentication-Info',
-            headers['Access-Control-Expose-Headers']
-        )
+        try:
+            conn_args = self.get_connect_args()
+            url = f'https://{conn_args["host"]}:{conn_args["port"]}/auth/token'
+
+            req = urllib.request.Request(url, method='OPTIONS')
+            req.add_header('Origin', 'https://example.edgedb.com')
+            response = urllib.request.urlopen(
+                req, context=self.tls_context
+            )
+            self.assertNotIn('Access-Control-Allow-Origin', response.headers)
+
+            await self.con.execute(
+                'configure instance '
+                'set cors_allow_origins := {"https://example.edgedb.com"}')
+            await self._wait_for_db_config(
+                'cors_allow_origins', instance_config=True)
+
+            req = urllib.request.Request(url, method='OPTIONS')
+            req.add_header('Origin', 'https://example.edgedb.com')
+            response = urllib.request.urlopen(
+                req, context=self.tls_context
+            )
+            headers = response.headers
 
-        with self.http_con() as con:
-            _, headers, status = self.http_con_request(
-                con, {}, path="token",
-                headers={'Origin': 'https://example.edgedb.com'}
+            self.assertIn('Access-Control-Allow-Origin', headers)
+            self.assertEqual(
+                headers['Access-Control-Allow-Origin'],
+                'https://example.edgedb.com'
+            )
+            self.assertIn('GET', headers['Access-Control-Allow-Methods'])
+            self.assertIn(
+                'Authorization',
+                headers['Access-Control-Allow-Headers']
             )
-            self.assertEqual(status, 401)
-            self.assertIn('access-control-allow-origin', headers)
-            self.assertIn('access-control-expose-headers', headers)
             self.assertIn(
                 'WWW-Authenticate',
-                headers['access-control-expose-headers']
+                headers['Access-Control-Expose-Headers']
             )
             self.assertIn(
                 'Authentication-Info',
-                headers['access-control-expose-headers']
+                headers['Access-Control-Expose-Headers']
             )
 
+            with self.http_con() as con:
+                _, headers, status = self.http_con_request(
+                    con, {}, path="token",
+                    headers={'Origin': 'https://example.edgedb.com'}
+                )
```

Comment:
Since we're in here, is there a reason we use `urllib.request` in the first case, but `http_con` here?